Sources/NIOSSL/TLSConfiguration.swift

```diff
@@ -62,6 +62,23 @@ public enum CertificateVerification {
     case fullVerification
 }
 
+/// Support for TLS renegotiation.
+///
+/// In general, renegotiation should not be enabled except in circumstances where it is absolutely necessary.
+/// Renegotiation is only supported in TLS 1.2 and earlier, and generally does not work very well. NIOSSL will
+/// disallow most uses of renegotiation: the only supported use-case is to perform post-connection authentication
+/// *as a client*. There is no way to initiate a TLS renegotiation in NIOSSL.
+public enum NIORenegotiationSupport {
```

Comment:
suggestions: Couldn't we nest this type in `TLSConfiguration` so ie would be `TLSConfiguration.RenegotiationSupport`? Then it doesn't need the `NIO` prefix either

Reply:
The only reason I put the type here was to keep it in line with the other types used by `TLSConfiguration`, which are at the top level. If you think that consistency is less important than the convenience of nesting the type, I'm more than happy to move it: I don't really care either way.

Reply:
@Lukasa I really don't mind either. Your call!

Reply:
Let's leave it where it is: status quo trumps convenience when it comes to the number of types we're discussing. I'll file an issue to move them all into the TLSConfiguration struct for v3.